integration-tests/src/test/java/oracle/kubernetes/operator/utils/Domain.java

```diff
@@ -248,6 +249,57 @@ public void verifyAdminServerExternalService(String username, String password) t
     }
   }
 
+  /**
+   * Verify that we have the channel set for the cluster
+   *
+   * @param protocol
+   * @param port
+   * @param path
+   * @throws Exception
+   */
+  public void verifyHasClusterServiceChannelPort(String protocol, int port, String path)
+      throws Exception {
+
+    /* Make sure the service exists in k8s */
+    if (!TestUtils.checkHasServiceChannelPort(
+        this.getDomainUid() + "-cluster-" + this.clusterName, domainNS, protocol, port)) {
+      throw new RuntimeException(
+          "FAILURE: Cannot find channel port in cluster, but expecting one: "
+              + port
+              + "/"
+              + protocol);
+    }
+
+    /* Make sure we can reach the port */
+    StringBuffer testAppUrl = new StringBuffer("http://");
+    testAppUrl
+        .append(this.getDomainUid() + "-cluster-" + this.clusterName)
+        .append(":")
+        .append(port)
+        .append("/");
+    testAppUrl.append(path);
+    // curl cmd to call webapp
+    StringBuffer curlCmd =
+        new StringBuffer(
+            "kubectl exec " + this.getDomainUid() + "-" + this.adminServerName + " /usr/bin/curl ");
+    curlCmd.append(testAppUrl.toString());
+
+    // curl cmd to get response code
+    StringBuffer curlCmdResCode = new StringBuffer(curlCmd.toString());
+    //  -- means pass trailing args to command in exec, not exec itself
+    curlCmdResCode.append(" -- --write-out %{http_code} -o /dev/null");
+
+    logger.info("Curl cmd with response code " + curlCmdResCode);
+    logger.info("Curl cmd " + curlCmd);
+
+    // call webapp iteratively till its deployed/ready
+    callWebAppAndWaitTillReady(curlCmdResCode.toString());
+
+    // execute curl and look for the managed server name in response
+    callWebAppAndCheckForServerNameInResponse(curlCmd.toString(), true);
```

Comment:
do u need to verify load balancing of requests here? if not, you can set false.

Reply:
I.e. we pass along the URL to the cluster, which resolves in round robin to each managed server in turn.